packages/amplify-dotnet-function-runtime-provider/src/utils/invoke.ts

```diff
@@ -1,82 +1,58 @@
 import path from 'path';
 import fs from 'fs-extra';
 import glob from 'glob';
-import childProcess from 'child_process';
-import { constants } from '../constants';
+import * as execa from 'execa';
 import { InvocationRequest } from 'amplify-function-plugin-interface';
 import { build } from './build';
+import { shimSrcPath, shimExecutablePath, executableName, shimBinPath } from '../constants';
 
-export async function invoke(request: InvocationRequest): Promise<string> {
-  if (await isInvokerStale(constants.shimSrcPath, constants.shimExecutablePath)) {
+export const invoke = async (request: InvocationRequest): Promise<string> => {
+  if (await isInvokerStale(shimSrcPath, shimExecutablePath)) {
     await buildInvoker();
   }
+
   await build({
     env: request.env,
     runtime: request.runtime,
     srcRoot: request.srcRoot,
     lastBuildTimestamp: request.lastBuildTimestamp,
   });
 
-  const distPath = path.join(request.srcRoot, 'dist');
-  return new Promise<string>((resolve, reject) => {
-    const invokeCommand = childProcess.spawn('dotnet', [constants.shimExecutablePath, request.handler], {
-      cwd: request.srcRoot,
-      env: {
-        ...process.env,
-        ...request.envVars,
-      },
-      stdio: ['pipe', 'pipe', process.stderr],
-    });
-
-    invokeCommand.stdin.setDefaultEncoding('utf-8');
-    invokeCommand.stdin.write(request.event);
-    invokeCommand.stdin.end();
+  const result = execa.sync(executableName, [shimExecutablePath, request.handler], {
+    cwd: request.srcRoot,
+    env: {
+      ...process.env,
```

Comment:
do we want to be passing the cli's env to the lambda?

Reply:
you can have extra things there, it is fine. no harm can be made.